content/articles/building-a-multiclass-image-classification-model-using-mobilenet-v2-and-tensorflow/index.md

```diff
@@ -0,0 +1,363 @@
+
+We will download the dataset from `tensorflow_datasets` `tensorflow_datasets` is an open-source repository of datasets for object detection, object segmentation, image tasks, video tasks, and natural language processing.
+
+Let's import `tensorflow_datasets`. 
+
+```python
+import tensorflow_datasets as tfds
+```
+We download the beans dataset as follows:
+
+```python
+beans_dataset, beans_info = tfds.load(name='beans', with_info=True, as_supervised=True, split=['train','test','validation'])
+```
+From the code snippet above, the `load` function will load the `beans` dataset. `with_info=True` will show the beans dataset information and metadata. `as_supervised=True` specifies the type of machine learning. We are dealing with supervised machine learning because the beans dataset is labeled. `split` splits the dataset into three sets. The sets are for model training, testing, and validation.
+
+To check the beans dataset, use this code:
+
+```python
+beans_info
+```
+The code produces this output:
+
+![Beans dataset](/engineering-education/building-a-multiclass-image-classification-model-using-mobilenet-v2-and-tensorflow/beans-info.png)
+
+The beans dataset contain images of bean leaves taken using mobile phones. It has 3 classes. 2 are beans disease classes and one healthy bean leaves. The beans disease classes are `Angular Leaf Spot` and `Bean Rust`.
+
+The beans dataset has 1295 images. After splitting, the test images are 128, train 1034, and validation 133. 
+
+### Showing the beans images
+We will display some of the images using the following code:
+
+```python
+train, info_train = tfds.load(name='beans', with_info=True, split='test')
+tfds.show_examples(info_train,train)
+```
+The code will display some of the train images as shown below:
+
+![Train images](/engineering-education/building-a-multiclass-image-classification-model-using-mobilenet-v2-and-tensorflow/train-images.png)
+
+From this output, `Angular Leaf Spot` leaves are labeled 0, `Bean Rust` 1, and `healthy` 2. 
+
+### Image preprocessing
+Image preprocessing will convert the beans image dataset into a format the neural network can use. It involves various stages. Lets create a function that will perform [image scaling](https://en.wikipedia.org/wiki/Image_scaling), [normalization](https://en.wikipedia.org/wiki/Normalization_(image_processing)) and [one hot encoding](https://en.wikipedia.org/wiki/One-hot).
+
+#### Function for image preprocessing
+We create this function using this code:
+
+```python
+def preprocessing(image, label):
+  image /= 255.0
+
+  return tf.image.resize(image,[224,224]), tf.one_hot(label, 3)
+```
+Image normalization converts the output pixel value between 0 by 1. The `preprocessing` function performs `normalization` by dividing the `image` by 255. The function uses ` tf.image.resize` to resize the image to `224,224`. It is the image size that MobileNetV2 expects.
```

Comment:
Use ` `` ` for keywords, variables, functions or anything that you want to be highlighted.

And, remove any additional spaces